src/actions/sso/authenticationRequest.ts

```diff
@@ -21,10 +23,23 @@ export const consumeAuthenticationRequest = (
   authenticationRequest: JSONWebToken<Authentication>,
   isDeepLinkInteraction: boolean = false,
 ): ThunkAction => async (dispatch, getState, backendMiddleware) => {
-  const { identityWallet } = backendMiddleware
+  const { identityWallet, registry } = backendMiddleware
   await identityWallet.validateJWT(authenticationRequest)
+  const { did: requesterDid, publicProfile } = await registry.resolve(
```

Comment:
Note that `registry.resolve` would only resolve DIDs from the jolocom registry, by definition. I think the BackendMiddleware should probably have a `resolveDID` method (which should be able to resolve any DID), if that needs to be used in the wallet code itself. But at the same time, the `publicProfile` is basically jolocom specific

Reply:
Ideally the `publicProfile` would not be required to be a `SignedCredential` but rather something that allows for multiple compliant implementations. It could be part of the resolution function to handle different public profile representations (used by different DID Methods) and bring them to a standard form usable by the wallet (i.e. a JSON object on the `Identity` Class).

Definitely outside of the scope of this issue I'd say, since it's only purpose is to make sure we are consistent in our logic across the different interaction screens.